### DISCUSSION TITLE
Positive and negative marker genes for integrated data

### DISCUSSION THREAD

**Discussion by dahun73** [satijalab/seurat#4636] [Q&A] [answered]
Hello all,

I have a question about marker genes. When I get marker genes using "FindAllMarkers", there are two types of log2FC, positive and negative. In this case, when I combined two samples such as control and stim, are these marker genes from comparing with all other cells? Or, from comparing with two different condition? I don't know exactly about these log2FC values whether these are from comparing with entire cells or not. I want to check out differentially expressed genes between two samples each cluster. 

I'm sorry for confusing question.
Thank you for reading

**Comment by saketkc** [comment, marked as answer]
`FindAllMarkers` takes each ident and compares it with all remaining idents. If you are looking for only positive markers, you can pass `only.pos=T`.

**Reply by saketkc** [reply]
Also, `FindAllMarkers` should be run on the RNA/SCT assay and not the integrated assay.

**Reply by dahun73** [reply]
Thank you so much :)
I can understand meaning of markers.

You said "FindAllMarkers should be run on the RNA/SCT assay and not the integrated assay.". If so, `FindAllMarkers` should be run separately such as control and stim, right..? Or, you mean I should on the RNA assay in seurat object?

I confused whether I can use `FindAllMarkers` after integration of two data.

**Reply by joseale2310** [reply]
@dahun73 No, you need to revert back to the RNA assay to use `FindAllMarkers`. Integration is useful for visualization.

**Reply by dahun73** [reply]
Thank you for reply. I totally agree that integration is useful for visualization. 
Then, you suggest that I should run `FindAllMarkers` individually, right?
But I'm curious about running `FindAllMarkers` after integration. Because some of clusters have totally different cell numbers between control and case. In this case, I think those markers in the integrated data may be meaningful.
Also, if I run the `FindAllMarkers` with each sample, how can I compare with those markers between control and case? I think it is hard to compare side-by-side which genes are significantly different pattern.

**Reply by joseale2310** [reply]
Check the basic [integration tutorial](https://satijalab.org/seurat/articles/integration_introduction.html). After integration, revert back to the RNA Assay and perform DEA there.

`DefaultAssay(immune.combined) <- "RNA"`
`nk.markers <- FindConservedMarkers(immune.combined, ident.1 = 6, grouping.var = "stim", verbose = FALSE)`
`head(nk.markers)`

It says `FindConservedMarkers` but you can run `FindAllMarkers` instead. I really do not think you should use the markers from the integrated data, since the underlaying data has been modified for the integration.
